Sources/NIOPosix/PosixSingletons+ConcurrencyTakeOver.swift

```diff
@@ -0,0 +1,91 @@
+import Atomics
+import NIOCore
+
+private let _haveWeTakenOverTheConcurrencyPool = ManagedAtomic(false)
+extension NIOSingletons {
+    /// Install ``MultiThreadedEventLoopGroup/singleton`` as Swift Concurrency's global executor.
+    ///
+    /// This allows to use Swift Concurrency and retain high-performance I/O alleviating the otherwise necessary thread switches between
+    /// Swift Concurrency's own global pool and a place (like an `EventLoop`) that allows to perform I/O
+    ///
+    /// This method uses an atomic compare and exchange to install the hook (and makes sure it's not already set). This unilateral atomic memory
+    /// operation doesn't guarantee anything becuase another piece of code could have done the same without using atomic operations. But we
```

Comment:
```suggestion
    /// operation doesn't guarantee anything because another piece of code could have done the same without using atomic operations. But we
```